### PR TITLE
LVFS HPAC for Dongle/24G device

### PR DESCRIPTION
**Pull request by sam412081go**
1. Fix bug of "ioctl error:broken pipe", the param of len is wrong in "fu_pxi_composite_receiver_cmd". (fu-pxi-common.c)

 2. Remove get response in "fu_pxi_receiver_device_write_payload", don't need it. Device won't response every payload. (fu-pxi-receiver-device.c)

 3. Implement HPAC OTA to fu-pxi-receiver-device.c

 4. Implement HPAC OTA to fu-pxi-wireless-device.c

 5. Modify OTA flow and mehtod for fu-pix-wireless-device :
    * Add "FW_OTA_PRECEDING" cmd at first of OTA.
    * Modify "fu_pxi_wireless_device_check_crc" cmd target to "FU_PXI_WIRELESS_DEVICE_TARGET_RECEIVER". * Add response check in "object_create", "init_new", and "fw_upgrade" function. * Remove response check in "write_payload" function. * Modify checksum caculation method in "write_chunk" function to "fu-pxi-receiver-device.c" and "fu-pxi-wireless-device.c". * Add idx param to "fu_pxi_receiver_device_get_peripheral_info" function.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [*] Code fix
- [* ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by sam412081go** [comment]
@superm1 @hughsie 
The latest commit(91e01b4178e5956ac2fe9ce183cabc10410914f3) will cause compile error.

[226/795] Compiling C object plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o
FAILED: plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o 
ccache cc -Iplugins/amd-gpu/libfu_plugin_amd_gpu.a.p -Iplugins/amd-gpu -I../plugins/amd-gpu -I. -I.. -Ilibfwupd -I../libfwupd -Ilibfwupdplugin -I../libfwupdplugin -Isubprojects/libxmlb/src -I../subprojects/libxmlb/src -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/p11-kit-1 -I/usr/include/gudev-1.0 -I/usr/local/include/gusb-1 -I/usr/include/libusb-1.0 -I/usr/include/json-glib-1.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -Waggregate-return -Wunused -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wdiscarded-qualifiers -Wduplicated-branches -Wduplicated-cond -Wempty-body -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Wimplicit-int -Winit-self -Wint-conversion -Wlogical-op -Wmaybe-uninitialized -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-cast-function-type -Wno-address-of-packed-member -Wno-unknown-pragmas -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wold-style-definition -Woverride-init -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wunused-variable -Wvla -Wwrite-strings -fstack-protector-strong -DFWUPD_COMPILATION -D_DEFAULT_SOURCE -D_BSD_SOURCE -D__BSD_VISIBLE -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -D_DARWIN_C_SOURCE=900000 -fPIC -pthread '-DG_LOG_DOMAIN="FuPluginAmdGpu"' -MD -MQ plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o -MF plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o.d -o plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o -c ../plugins/amd-gpu/fu-amd-gpu-device.c
../plugins/amd-gpu/fu-amd-gpu-device.c: In function ‘fu_amd_gpu_device_setup’:
../plugins/amd-gpu/fu-amd-gpu-device.c:111:31: error: storage size of ‘vbios_info’ isn’t known
  111 |  struct drm_amdgpu_info_vbios vbios_info;
      |                               ^~~~~~~~~~
../plugins/amd-gpu/fu-amd-gpu-device.c:115:28: error: invalid application of ‘sizeof’ to incomplete type ‘struct drm_amdgpu_info_vbios’
  115 |      .return_size = sizeof(struct drm_amdgpu_info_vbios),
      |                            ^~~~~~
../plugins/amd-gpu/fu-amd-gpu-device.c:116:25: error: ‘AMDGPU_INFO_VBIOS_INFO’ undeclared (first use in this function); did you mean ‘AMDGPU_INFO_VBIOS_IMAGE’?
  116 |      .vbios_info.type = AMDGPU_INFO_VBIOS_INFO,
      |                         ^~~~~~~~~~~~~~~~~~~~~~
      |                         AMDGPU_INFO_VBIOS_IMAGE
../plugins/amd-gpu/fu-amd-gpu-device.c:116:25: note: each undeclared identifier is reported only once for each function it appears in
../plugins/amd-gpu/fu-amd-gpu-device.c:111:31: warning: unused variable ‘vbios_info’ [-Wunused-variable]
  111 |  struct drm_amdgpu_info_vbios vbios_info;
      |                               ^~~~~~~~~~
[235/795] Compiling C object plugins/ccgx/libfu_plugin_ccgx.a.p/fu-ccgx-plugin.c.o
ninja: build stopped: subcommand failed.

**Comment by superm1** [comment]
> @superm1 @hughsie The latest commit([91e01b4](https://github.com/fwupd/fwupd/commit/91e01b4178e5956ac2fe9ce183cabc10410914f3)) will cause compile error.
> 
> [226/795] Compiling C object plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o FAILED: plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o ccache cc -Iplugins/amd-gpu/libfu_plugin_amd_gpu.a.p -Iplugins/amd-gpu -I../plugins/amd-gpu -I. -I.. -Ilibfwupd -I../libfwupd -Ilibfwupdplugin -I../libfwupdplugin -Isubprojects/libxmlb/src -I../subprojects/libxmlb/src -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/gio-unix-2.0 -I/usr/include/p11-kit-1 -I/usr/include/gudev-1.0 -I/usr/local/include/gusb-1 -I/usr/include/libusb-1.0 -I/usr/include/json-glib-1.0 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=c11 -O0 -g -Waggregate-return -Wunused -Warray-bounds -Wcast-align -Wclobbered -Wdeclaration-after-statement -Wdiscarded-qualifiers -Wduplicated-branches -Wduplicated-cond -Wempty-body -Wformat=2 -Wformat-nonliteral -Wformat-security -Wformat-signedness -Wignored-qualifiers -Wimplicit-function-declaration -Wimplicit-int -Winit-self -Wint-conversion -Wlogical-op -Wmaybe-uninitialized -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-cast-function-type -Wno-address-of-packed-member -Wno-unknown-pragmas -Wno-missing-field-initializers -Wno-strict-aliasing -Wno-suggest-attribute=format -Wno-unused-parameter -Wold-style-definition -Woverride-init -Wpointer-arith -Wredundant-decls -Wreturn-type -Wshadow -Wsign-compare -Wstrict-aliasing -Wstrict-prototypes -Wswitch-default -Wtype-limits -Wundef -Wuninitialized -Wunused-but-set-variable -Wunused-variable -Wvla -Wwrite-strings -fstack-protector-strong -DFWUPD_COMPILATION -D_DEFAULT_SOURCE -D_BSD_SOURCE -D__BSD_VISIBLE -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -D_DARWIN_C_SOURCE=900000 -fPIC -pthread '-DG_LOG_DOMAIN="FuPluginAmdGpu"' -MD -MQ plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o -MF plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o.d -o plugins/amd-gpu/libfu_plugin_amd_gpu.a.p/fu-amd-gpu-device.c.o -c ../plugins/amd-gpu/fu-amd-gpu-device.c ../plugins/amd-gpu/fu-amd-gpu-device.c: In function ‘fu_amd_gpu_device_setup’: ../plugins/amd-gpu/fu-amd-gpu-device.c:111:31: error: storage size of ‘vbios_info’ isn’t known 111 | struct drm_amdgpu_info_vbios vbios_info; | ^~~~~~~~~~ ../plugins/amd-gpu/fu-amd-gpu-device.c:115:28: error: invalid application of ‘sizeof’ to incomplete type ‘struct drm_amdgpu_info_vbios’ 115 | .return_size = sizeof(struct drm_amdgpu_info_vbios), | ^~~~~~ ../plugins/amd-gpu/fu-amd-gpu-device.c:116:25: error: ‘AMDGPU_INFO_VBIOS_INFO’ undeclared (first use in this function); did you mean ‘AMDGPU_INFO_VBIOS_IMAGE’? 116 | .vbios_info.type = AMDGPU_INFO_VBIOS_INFO, | ^~~~~~~~~~~~~~~~~~~~~~ | AMDGPU_INFO_VBIOS_IMAGE ../plugins/amd-gpu/fu-amd-gpu-device.c:116:25: note: each undeclared identifier is reported only once for each function it appears in ../plugins/amd-gpu/fu-amd-gpu-device.c:111:31: warning: unused variable ‘vbios_info’ [-Wunused-variable] 111 | struct drm_amdgpu_info_vbios vbios_info; | ^~~~~~~~~~ [235/795] Compiling C object plugins/ccgx/libfu_plugin_ccgx.a.p/fu-ccgx-plugin.c.o ninja: build stopped: subcommand failed.

Can you share more about the environment you get this build failure?  CI doesn't hit it and I'd like to reproduce it.

**Comment by sam412081go** [comment]
@superm1 

 My OS version
Description:	Ubuntu 20.04.5 LTS
Release:	20.04

msg of using 'env' cmd.

sdrf@OA-109048:~/fwupd_fw2_20231004/fwupd/build$ env
SHELL=/bin/bash
SESSION_MANAGER=local/OA-109048:@/tmp/.ICE-unix/1712,unix/OA-109048:/tmp/.ICE-unix/1712
QT_ACCESSIBILITY=1
COLORTERM=truecolor
XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg
XDG_MENU_PREFIX=gnome-
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
LC_ADDRESS=lzh_TW
GNOME_SHELL_SESSION_MODE=ubuntu
LC_NAME=lzh_TW
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
XMODIFIERS=@im=ibus
DESKTOP_SESSION=ubuntu
LC_MONETARY=lzh_TW
SSH_AGENT_PID=1674
GTK_MODULES=gail:atk-bridge
PWD=/home/sdrf/fwupd_fw2_20231004/fwupd/build
LOGNAME=sdrf
XDG_SESSION_DESKTOP=ubuntu
XDG_SESSION_TYPE=x11
GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1
XAUTHORITY=/run/user/1000/gdm/Xauthority
GJS_DEBUG_TOPICS=JS ERROR;JS LOG
WINDOWPATH=2
HOME=/home/sdrf
USERNAME=sdrf
IM_CONFIG_PHASE=1
LC_PAPER=lzh_TW
LANG=en_US.UTF-8
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
XDG_CURRENT_DESKTOP=ubuntu:GNOME
VTE_VERSION=6003
GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/17bb3e89_8fc9_4064_a727_d3ab2d753160
INVOCATION_ID=ac3dadd9880f4930b13ecbefd0a5ea35
MANAGERPID=1497
GJS_DEBUG_OUTPUT=stderr
LESSCLOSE=/usr/bin/lesspipe %s %s
XDG_SESSION_CLASS=user
TERM=xterm-256color
LC_IDENTIFICATION=lzh_TW
LESSOPEN=| /usr/bin/lesspipe %s
USER=sdrf
GNOME_TERMINAL_SERVICE=:1.82
DISPLAY=:0
SHLVL=1
LC_TELEPHONE=lzh_TW
QT_IM_MODULE=ibus
LC_MEASUREMENT=lzh_TW
XDG_RUNTIME_DIR=/run/user/1000
LC_TIME=lzh_TW
JOURNAL_STREAM=8:47594
XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
PATH=/home/sdrf/.local/bin:/usr/local/bin:/usr/lib/gcc/gcc-arm-none-eabi-10.3-2021.10/bin/:/home/sdrf/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
GDMSESSION=ubuntu
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
LC_NUMERIC=lzh_TW
OLDPWD=/home/sdrf
_=/usr/bin/env

**Comment by superm1** [comment]
> @superm1
> 
> My OS version Description: Ubuntu 20.04.5 LTS Release: 20.04
> 
> msg of using 'env' cmd.
> 
> sdrf@OA-109048:~/fwupd_fw2_20231004/fwupd/build$ env SHELL=/bin/bash SESSION_MANAGER=local/OA-109048:@/tmp/.ICE-unix/1712,unix/OA-109048:/tmp/.ICE-unix/1712 QT_ACCESSIBILITY=1 COLORTERM=truecolor XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg XDG_MENU_PREFIX=gnome- GNOME_DESKTOP_SESSION_ID=this-is-deprecated LC_ADDRESS=lzh_TW GNOME_SHELL_SESSION_MODE=ubuntu LC_NAME=lzh_TW SSH_AUTH_SOCK=/run/user/1000/keyring/ssh XMODIFIERS=@im=ibus DESKTOP_SESSION=ubuntu LC_MONETARY=lzh_TW SSH_AGENT_PID=1674 GTK_MODULES=gail:atk-bridge PWD=/home/sdrf/fwupd_fw2_20231004/fwupd/build LOGNAME=sdrf XDG_SESSION_DESKTOP=ubuntu XDG_SESSION_TYPE=x11 GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1 XAUTHORITY=/run/user/1000/gdm/Xauthority GJS_DEBUG_TOPICS=JS ERROR;JS LOG WINDOWPATH=2 HOME=/home/sdrf USERNAME=sdrf IM_CONFIG_PHASE=1 LC_PAPER=lzh_TW LANG=en_US.UTF-8 LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:_.tar=01;31:_.tgz=01;31:_.arc=01;31:_.arj=01;31:_.taz=01;31:_.lha=01;31:_.lz4=01;31:_.lzh=01;31:_.lzma=01;31:_.tlz=01;31:_.txz=01;31:_.tzo=01;31:_.t7z=01;31:_.zip=01;31:_.z=01;31:_.dz=01;31:_.gz=01;31:_.lrz=01;31:_.lz=01;31:_.lzo=01;31:_.xz=01;31:_.zst=01;31:_.tzst=01;31:_.bz2=01;31:_.bz=01;31:_.tbz=01;31:_.tbz2=01;31:_.tz=01;31:_.deb=01;31:_.rpm=01;31:_.jar=01;31:_.war=01;31:_.ear=01;31:_.sar=01;31:_.rar=01;31:_.alz=01;31:_.ace=01;31:_.zoo=01;31:_.cpio=01;31:_.7z=01;31:_.rz=01;31:_.cab=01;31:_.wim=01;31:_.swm=01;31:_.dwm=01;31:_.esd=01;31:_.jpg=01;35:_.jpeg=01;35:_.mjpg=01;35:_.mjpeg=01;35:_.gif=01;35:_.bmp=01;35:_.pbm=01;35:_.pgm=01;35:_.ppm=01;35:_.tga=01;35:_.xbm=01;35:_.xpm=01;35:_.tif=01;35:_.tiff=01;35:_.png=01;35:_.svg=01;35:_.svgz=01;35:_.mng=01;35:_.pcx=01;35:_.mov=01;35:_.mpg=01;35:_.mpeg=01;35:_.m2v=01;35:_.mkv=01;35:_.webm=01;35:_.ogm=01;35:_.mp4=01;35:_.m4v=01;35:_.mp4v=01;35:_.vob=01;35:_.qt=01;35:_.nuv=01;35:_.wmv=01;35:_.asf=01;35:_.rm=01;35:_.rmvb=01;35:_.flc=01;35:_.avi=01;35:_.fli=01;35:_.flv=01;35:_.gl=01;35:_.dl=01;35:_.xcf=01;35:_.xwd=01;35:_.yuv=01;35:_.cgm=01;35:_.emf=01;35:_.ogv=01;35:_.ogx=01;35:_.aac=00;36:_.au=00;36:_.flac=00;36:_.m4a=00;36:_.mid=00;36:_.midi=00;36:_.mka=00;36:_.mp3=00;36:_.mpc=00;36:_.ogg=00;36:_.ra=00;36:_.wav=00;36:_.oga=00;36:_.opus=00;36:_.spx=00;36:*.xspf=00;36: XDG_CURRENT_DESKTOP=ubuntu:GNOME VTE_VERSION=6003 GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/17bb3e89_8fc9_4064_a727_d3ab2d753160 INVOCATION_ID=ac3dadd9880f4930b13ecbefd0a5ea35 MANAGERPID=1497 GJS_DEBUG_OUTPUT=stderr LESSCLOSE=/usr/bin/lesspipe %s %s XDG_SESSION_CLASS=user TERM=xterm-256color LC_IDENTIFICATION=lzh_TW LESSOPEN=| /usr/bin/lesspipe %s USER=sdrf GNOME_TERMINAL_SERVICE=:1.82 DISPLAY=:0 SHLVL=1 LC_TELEPHONE=lzh_TW QT_IM_MODULE=ibus LC_MEASUREMENT=lzh_TW XDG_RUNTIME_DIR=/run/user/1000 LC_TIME=lzh_TW JOURNAL_STREAM=8:47594 XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop PATH=/home/sdrf/.local/bin:/usr/local/bin:/usr/lib/gcc/gcc-arm-none-eabi-10.3-2021.10/bin/:/home/sdrf/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin GDMSESSION=ubuntu DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus LC_NUMERIC=lzh_TW OLDPWD=/home/sdrf _=/usr/bin/env

OK, so it's on older Ubuntu.  I'll see if I can reproduce in a container, thanks!

For now you can `-Dplugin_amdgpu=disabled` in meson.

**Comment by sam412081go** [comment]
@superm1  got it, thank you!

File: plugins/pixart-rf/fu-pxi-receiver-device.c [hunk 1]
```diff
@@ -601,16 +627,19 @@ fu_pxi_receiver_device_write_firmware(FuDevice *device,
...
 	guint8 buf[FU_PXI_RECEIVER_DEVICE_OTA_BUF_SZ] = {0x0};
 	guint16 checksum = 0;
+	guint16 hpac_ver = 0;
 	g_autofree gchar *version_str = NULL;
 	g_autoptr(GByteArray) ota_cmd = g_byte_array_new();
 	g_autoptr(GByteArray) receiver_device_cmd = g_byte_array_new();
 
 	fu_byte_array_append_uint8(ota_cmd, 0x1); /* ota init new command length */
 	fu_byte_array_append_uint8(ota_cmd, FU_PXI_DEVICE_CMD_FW_OTA_GET_MODEL);
+	fu_byte_array_append_uint8(ota_cmd, idx);
```
Comment (hughsie) [inline plugins/pixart-rf/fu-pxi-receiver-device.c:642]: isn't this an API break? i.e. how will older devices know there's an index here?

Comment (sam412081go) [reply plugins/pixart-rf/fu-pxi-receiver-device.c:642]: We originally reserved this parameter in firmware, which was not utilized by older devices. Now, the 'idx' parameter is being used, so it doesn't affect OTA for older devices.

**Pull request merged**